src/components/select/select.component.ts

```diff
@@ -151,6 +151,12 @@ export class SelectComponent<T> implements OnInit, OnChanges, OnDestroy, Control
      */
     @Input() optionTemplate: TemplateRef<any>;
 
+    /** Container for saving the recently selected options. */
+    @Input() recentOptions: T[];

```

Comment:
Can we use the `ReadonlyArray<T>` type for the new inputs and outputs you added instead of standard array types `T[]`. 

Enforcing immutability can prevent issues with change detection, as change detection performs a `===` comparison, so allow things like `push` to the array can prevent change detection from noticing the change, whereas the `ReadonlyArray` type removes any functions that will change the array without changing the array instance.

Reply:
We have a use case where the options can be changed after initial setup. That is when two drop-down lists on the same time share the history. Therefore, the array should not be immutable.

Reply:
The `ReadonlyArray` type doesn't mean that it can't be changed be from the outside, it just means that the array reference needs changed.

So you can still pass it new values at any time, it just makes sure that any time you want to change the array contents you must also change the reference.

Arrays normally have functions like `push` which will change the contents of the array, however if we are to compare the old value with the new value:

```typescript
const oldValue = myArray;
const newValue = myArray;

newValue.push('Some Value');

newValue === oldValue // this will result in true, even though we have pushed to the array
```

Change detection in Angular uses a simple `===` compare to detect changes, so by using `push` to mutate the array the change might not be detected.

If we use a readonly array it enforces us to create a new array each time with the updated content, eg:

```typescript
const oldValue = myArray;
const newValue = [...myArray];

newValue = [...newValue, 'new value'];

newValue === oldValue // results in false
```

Therefore change detection can easily and very performantly detect the change.

Reply:
Done

Reply:
Thank you for your detailed explanations!